vignettes/independent_filtering_plots.Rmd

```diff
@@ -0,0 +1,200 @@
+n1 <- n2 <- 3
+set.seed(1969)
+use <- unlist(tapply(1:ncol(ALL_bcrneg),
+ALL_bcrneg$mol.biol, sample, n1))
+subsample <- ALL_bcrneg[,use]
+```
+
+We now use functions from *genefilter* to compute overall standard
+deviation filter statistics as well as standard two-sample $t$ and
+related statistics.
+
+```{r stats, cache=TRUE}
+S <- rowSds( exprs( subsample ) )
+temp <- rowttests( subsample, subsample$mol.biol )
+d <- temp$dm
+p <- temp$p.value
+t <- temp$statistic
+```
+
+
+# Filtering volcano plot
+
+Filtering on overall standard deviation and then using a standard
+$t$-statistic induces a lower bound of fold change, albeit one which
+varies somewhat with the significance of the $t$-statistic. The
+`filter_volcano` function allows you to visualize this effect.
+
+The output is shown in the left panel of Fig. \@ref(fig:filter-volcano).
+
+The `kappa_p` and `kappa_t` functions, used to make the volcano plot,
+compute the fold change bound multiplier as a function of either a
+$t$-test $p$-value or the $t$-statistic itself. The actual induced bound
+on the fold change is $\kappa$ times the filter's cutoff on the overall
+standard deviation. Note that fold change bounds for values of $|T|$
+which are close to 0 are not of practical interest because we will not
+reject the null hypothesis with test statistics in this range.
+
+The output is shown in the right panel of Fig. \@ref(fig:filter-volcano).
+
+
+```{r filter-volcano, fig.cap = "Left panel: plot produced by the filter_volcano function.  Right panel: graph of the `kappa_t` function."}
+par(mfrow=c(1, 2))
+
+S_cutoff <- quantile(S, .50)
+ss <- filter_volcano(d, p, S, n1, n2, alpha=.01, S_cutoff)
+
+t <- seq(0, 5, length=100)
+plot(t, kappa_t(t, n1, n2) * S_cutoff,
+     xlab="|T|", ylab="Fold change bound", type="l")
+```
+
+# Rejection count plots
+
+## Across $p$-value cutoffs
+
+The `filtered_p` function permits easy simultaneous calculation of
+unadjusted or adjusted $p$-values over a range of filtering thresholds
+($\theta$). Here, we return to the full "BCR/ABL" versus "NEG" data set,
+and compute adjusted $p$-values using the method of Benjamini and
+Hochberg, for a range of different filter stringencies.
+
+```{r table}
+table(ALL_bcrneg$mol.biol)
+```
+
+```{r filtered_p, message=FALSE, warning=FALSE, include=TRUE}
+S2 <- rowVars(exprs(ALL_bcrneg))
+p2 <- rowttests(ALL_bcrneg, "mol.biol")$p.value
+theta <- seq(0, .5, .1)
+p_bh <- filtered_p(S2, p2, theta, method="BH")
+```
+
+```{r p_bh, echo=TRUE}
+head(p_bh)
+```
+
+The `rejection_plot` function takes sets of $p$-values corresponding to
+different filtering choices --- in the columns of a matrix or in a list
+--- and shows how rejection count ($R$) relates to the choice of cutoff
+for the $p$-values. For these data, over a reasonable range of FDR
+cutoffs, increased filtering corresponds to increased rejections.
+
+```{r rejection-plot, fig.cap = "Plot produced by the `rejection_plot function`"}
```

Comment:
In the vignette PDF, I see that "Figure 2" is in bold black font and the caption is black font whereas this caption has bold blue font as the caption text. Is this an easy edit? Not sure whether it would be a headache to match the original vignette PDF or not.

Reply:
We might be able to accomplish this with the MiKTeX package, but do we want to have another library?